pkg/kapp/diff/change_set_with_versioned_rs.go

```diff
@@ -76,19 +77,24 @@ func (d ChangeSetWithVersionedRs) Calculate() ([]Change, error) {
 	return allChanges, nil
 }
 
+func (d ChangeSetWithVersionedRs) versionedResourceName(res ctlres.Resource) VersionedResource {
```

Comment:
seems i forgot tbe TODO comment
this is my "splitting the overloaded VersionedResourxe type"  refactoring-quickfix.  the "name" suffix should qualify that it can only be use for name-related stuff, but not for reference-update stuff; for the latter the templateRules are missing.

later this will return an instance of the type dedicated to the name-related logic, properly renamed.